src/common/utils.py

```diff
@@ -389,3 +401,31 @@ def retrieve_max_cluster_size(region, proxy_config, asg_name, fallback):
         )
         log.critical(error_msg)
         raise CriticalError(error_msg)
+
+
+def get_cluster_instance_info(stack_name, region, proxy_config, include_master=False):
+    """Return a list of instance_id that are in the cluster."""
+    try:
+        instances_in_cluster = []
+        ec2_client = boto3.client("ec2", region_name=region, config=proxy_config)
+        instance_paginator = ec2_client.get_paginator("describe_instances")
+        nodes_to_include = ["Compute", "Master"] if include_master else ["Compute"]
+
+        for page in instance_paginator.paginate(
+            Filters=[
+                {"Name": "tag:Application", "Values": [stack_name]},
+                {"Name": "tag:Name", "Values": nodes_to_include},
+            ]
+        ):
+            for reservation in page.get("Reservations"):
```

Comment:
you want to double-check this cause I remember I was not getting Reservations but Instances directly when using the paginator. If I'm correct on this can we understand why this is not caught by integ tests?

Reply:
Response does include Reservations